src/Rs/Json/Patch/Document.php

```diff
@@ -78,7 +78,7 @@ private function extractPatchOperations($patchDocument)
      */
     private function isEmptyPatchDocument($patchDocument)
     {
-        return (empty($patchDocument) || !is_array($patchDocument) || count($patchDocument) === 0);
+        return (empty($patchDocument) || !is_array($patchDocument) || ( (!is_array($patchDocument)) || count($patchDocument) === 0));
```

Comment:
This seems overly complicated. Just empty($patchDocument) seems sufficient. Is there a reason for the extra conditions?

Reply:
Since this code comes from an external source I would prefer to not modify the logic just prevent the count error

Reply:
Is there evidence that the the original line is causing an error? Seems the only way the existing code could execute the count($patchDocument) === 0 is if $patchDocument is an array. count() works fine for array.